source/GameInterface/Services/ItemRosters/Handlers/Events/ItemRosterUpdateHandler.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Common.Messaging;
+using GameInterface.Services.ItemRosters.Messages.Events;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.Core;
+using TaleWorlds.CampaignSystem;
+using Serilog;
+using Common.Logging;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.ObjectSystem;
+using Common;
+using TaleWorlds.CampaignSystem.Roster;
+
+namespace GameInterface.Services.ItemRosters.Handlers.Events
+{
+    /// <summary>
+    /// Handles ItemRosterUpdated on client.
+    /// </summary>
+    internal class ItemRosterUpdateHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly ILogger logger;
+
+        public ItemRosterUpdateHandler(IMessageBroker messageBroker)
+        {
+            if (ModInformation.IsServer)
+                return;
+
+            this.messageBroker = messageBroker;
+
+            logger = LogManager.GetLogger<ItemRosterUpdateHandler>();
+
+            messageBroker.Subscribe<ItemRosterUpdate>(Handle);
+        }
+
+        public void Handle(MessagePayload<ItemRosterUpdate> payload)
+        {
+            GameLoopRunner.RunOnMainThread(() =>
+            {
+                ItemObject item;
+                ItemModifier modifier = null;
+                ItemRoster roster = null;
+
+                if ((item = MBObjectManager.Instance.GetObject<ItemObject>(payload.What.ItemID)) == null) {
```

Comment:
Use our IObjectManager instead, it may not have the functionality for ItemObject and ItemModifier though, let me know if that is the case

```suggestion
                if ((item = MBObjectManager.Instance.GetObject<ItemObject>(payload.What.ItemID)) == null) {
```